challenge-28/challenge-28.js

```diff
@@ -25,3 +27,152 @@
   - Utilize a lib DOM criada anteriormente para facilitar a manipulação e
   adicionar as informações em tela.
   */
+
+  function DOM (elements) {
+    this.element = doc.querySelectorAll(elements);
+  }
+
+  DOM.is = function (obj) {
+    return Object.prototype.toString.call(obj);
+  }
+
+  DOM.isArray = function (obj) {
+    return this.is(obj) === '[object Array]';
+  }
+
+  DOM.isObject = function (obj) {
+    return this.is(obj) === '[object Object]';
+  }
+
+  DOM.isFunction = function (obj) {
+    return this.is(obj) === '[object Function]';
+  }
+
+  DOM.isNumber = function (obj) {
+    return this.is(obj) === '[object Number]';
+  }
+
+  DOM.isString = function (obj) {
+    return this.is(obj) === '[object String]';
+  }
+
+  DOM.isBoolean = function (obj) {
+    return this.is(obj) === '[object Boolean]';
+  }
+
+  DOM.isNull = function (obj) {
+    return this.is(obj) === '[object Null]' || this.is(obj) === '[object Undefined]';
+  }
+
+  DOM.prototype.on = function on (event, callback) {
+    Array.prototype.forEach.call(this.element, function (element) {
+      element.addEventListener(event, callback, false);
+    });
+  }
+
+  DOM.prototype.off = function off (event, callback) {
+    Array.prototype.forEach.call(this.element, function (element) {
+      element.removeEventListener(event, callback, false);
+    });
+  }
+
+  DOM.prototype.get = function get () {
+    if (arguments === [])
```

Comment:
-1
Sabe que isso aqui não funciona né? Lembre-se que objetos tem uma referência específica na memória, logo não é possível comparar dois objetos diferentes :)

Outro problema é que aqui você está tentando comparar um objeto do tipo "Arguments" com um objeto do tipo "Array". De qualquer forma eles nunca seriam iguais :)

Lembre-se também que arguments é um "array-like", não um array de verdade xD

E mais um problema: se não passar argumentos, o retorno deveria ser o primeiro elemento de this.element, que é uma *Node List". Retornando this.element, você retorna a própria Node List, e vai ter que usar o índice [0], [1], etc., para se referir a um elemento específico :)

O que você pode fazer para validar aqui é testar arguments.length para saber se existe algum argumento passado.